src/osd/SnapMapper.cc

```diff
@@ -208,10 +208,14 @@ int OSDriver::get_next_or_current(
 string SnapMapper::get_prefix(int64_t pool, snapid_t snap)
 {
   static_assert(sizeof(pool) == 8, "assumed by the formatting code");
+
+  // note: the snap_id is to be formatted as a 64-bit hex number,
+  // and not according to the text representation of snapid_t
+  ceph_assert(snap != CEPH_NOSNAP && snap != CEPH_SNAPDIR);
   return fmt::sprintf("%s%lld_%.16X_",
 		      MAPPING_PREFIX,
 		      pool,
-		      snap);
+		      static_cast<uint64_t>(snap));
```

Comment:
Do we need a `Fixes` for that?

Reply:
I don't think so, since it could have been a bug for unpredictable users. For the existing ones, it should be ok or otherwise the assertion will catch it.

Reply:
CC @ronen-fr